crates/biome_js_analyze/src/lint/nursery/use_throw_only_error.rs

```diff
@@ -0,0 +1,141 @@
+use biome_analyze::{
+    context::RuleContext, declare_rule, Ast, Rule, RuleDiagnostic, RuleSource, RuleSourceKind,
+};
+use biome_console::markup;
+use biome_js_syntax::{
+    AnyJsExpression, AnyJsLiteralExpression, JsLogicalExpression, JsObjectExpression, JsSyntaxKind,
+    JsThrowStatement,
+};
+use biome_rowan::AstNode;
+
+declare_rule! {
+    /// Disallow throwing non-`Error` values.
+    ///
+    /// It is considered good practice to only throw the `Error` object itself or an object using the `Error` object
+    /// as base objects for user-defined exceptions. The fundamental benefit of `Error` objects is that they automatically
+    /// keep track of where they were built and originated.
+    ///
+    /// ## Examples
+    ///
+    /// ### Invalid
+    ///
+    /// ```js,expect_diagnostic
+    /// throw undefined;
+    /// ```
+    /// ```js,expect_diagnostic
+    /// throw false;
+    /// ```
+    /// ```js,expect_diagnostic
+    /// throw "a" + "b";
+    /// ```
+    ///
+    /// ### Valid
+    ///
+    /// ```js
+    /// throw new Error();
+    /// ```
+    /// ```js
+    /// throw new TypeError('biome');
+    /// ```
+    /// ```js
+    /// class CustomError extends Error {}
+    ///
+    /// throw new CustomError();
+    /// ```
+    ///
+    /// ## Caveats
+    ///
+    /// This rule only covers cases where throw value can be known statically,
+    /// complex cases such as object and function access are not checked yet.
+    /// This will be improved in the future, once Biome supports type inference.
```

Comment:
```suggestion
    /// This rule only covers cases where throwing the value can be known statically.
    /// Complex cases such as object and function access aren't checked.
    /// This will be improved in the future once Biome supports type inference.
```